flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/CachedBufferStorage.java

```diff
@@ -74,49 +102,139 @@ public void add(BufferOrEvent boe) {
 	}
 
 	/**
-	 * It is never reusing resources and is defaulting to {@link #rollOverWithoutReusingResources()}.
+	 * Starts a new sequence of buffers and event reusing the same resources and
+	 * returns the current sequence of buffers for reading.
+	 *
+	 * @return The readable sequence of buffers and events, or 'null', if nothing was added.
 	 */
-	@Override
 	public BufferOrEventSequence rollOverReusingResources() {
```

Comment:
It is no need to distinguish `rollOverReusingResources` and `rollOverWithoutReusingResources` if we only have `CachedBufferStorage`?  If so we can retain only one method, also for private method instead.